tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_browser_Browser.java

```diff
@@ -1105,6 +1105,35 @@ public void test_setText() {
 	validateTitleChanged(expectedTitle, browserSetFunc);
 }
 
+/**
+ * Corner-case, probably only relevant on Edge, see
+ * https://github.com/eclipse-platform/eclipse.platform.swt/pull/1463
+ */
+@Test
+public void test_setTextContainingScript_applicationLayerProgressListenerMustSeeUpToDateDom() {
+	AtomicReference<String> textAfterProgressCompletion = new AtomicReference<>();
+	browser.addProgressListener(ProgressListener.completedAdapter(event -> {
+		String script = """
+				var h1s = document.getElementsByTagName("h1");
+				h1s[0].innerHTML = "Hello, World!";
+				""";
+		browser.execute(script);
+		textAfterProgressCompletion.set(browser.getText());
+	}));
+	browser.setText("""
+			<html>
+				<head>
+					<script src=\"file:///does/not/really/needs/to/exist.js\"></script>
+				</head>
+				<body>
+					<h1>Initial Heading</h1>
+				</body>
+			</html>
+			""");
+	assertTrue(waitForPassCondition(() -> textAfterProgressCompletion.get() != null
+			&& textAfterProgressCompletion.get().contains("<h1>Hello, World!</h1>")));
```

Comment:
Not sure why the test is failing. Locally, it works fine (with both IE and Edge). Maybe improve the assertion to get better feedback? E.g. with the available Hamcrest matchers:
```suggestion
	assertThat(textAfterProgressCompletion.get(), containsString("<h1>Hello, World!</h1>"));
```

Reply:
Yeah same here on Windows. However, on Linux it did fail also locally, because getText() behaves differently.

Tried another approach to exfiltrate the modified DOM content.

Reply:
Finally all tests are green :)